src/query.ts

```diff
@@ -125,7 +125,7 @@ export class Query {
    * counter and set to empty the query term
    * @param {String[]}     items            The dfid list
    */
-  public addItems(items: string[]): void {
+  public searchItems(items: string[]): void {
```

Comment:
- Put this method after `search()`.
- Being purist, I'm not sure if we should change page here nor in `search()` because it's something that the user may not expect (although it ensures the right behavior, but it's dark magic).

Idea: regarding the already commented `kind` getter, it could be used inside `search()` and `searchItems` like this:

```js
public search(text: string): void {
  if (this.kind === SEARCH_ITEMS) {
    throw new QueryError(`can't set search terms in an items search, remove items first`);
  }
  // …
}

public searchItems(items: string[]): void {
  if (this.kind === SEARCH_TEXT) {
    throw new QueryError(`can't set search items in a text search, remove search text first`);
  }
  // …
}
```

So you add validation in `Query` and don't have to do black magic (change existing parameters when the dev didn't asked that).